src/Media_Command.php

```diff
@@ -693,6 +698,11 @@ private function needs_regeneration( $att_id, $fullsizepath, $is_pdf, $image_siz
 			$metadata['sizes'] = array( $image_size => $metadata['sizes'][ $image_size ] );
 		}
 
+		// In certain situations, eg if file uploaded when applicable image editor such as Imagick unavailable, the sizes metadata may not exist.
+		if ( ! isset( $metadata['sizes'] ) || ! is_array( $metadata['sizes'] ) ) {
+			$metadata['sizes'] = array();
+		}
+
 		if ( $this->image_sizes_differ( $image_sizes, $metadata['sizes'] ) ) {
```

Comment:
Do you think the above logical construction (the check to make sure `$metadata['sizes']` is an array) is useful for documentation purposes?

Otherwise, the code would be cleaner with a simple cast to array in the line where it is used...
```
if ( $this->image_sizes_differ( $image_sizes, (array) $metadata['sizes'] ) ) {
```